datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/PatchEntitiesResolver.java

```diff
@@ -0,0 +1,303 @@
+import com.linkedin.datahub.graphql.QueryContext;
+import com.linkedin.datahub.graphql.concurrency.GraphQLConcurrencyUtils;
+import com.linkedin.datahub.graphql.generated.PatchEntityInput;
+import com.linkedin.datahub.graphql.generated.PatchEntityResult;
+import com.linkedin.entity.client.EntityClient;
+import com.linkedin.metadata.entity.EntityService;
+import com.linkedin.metadata.entity.IngestResult;
+import com.linkedin.metadata.models.registry.EntityRegistry;
+import com.linkedin.metadata.utils.AuditStampUtils;
+import com.linkedin.metadata.utils.GenericRecordUtils;
+import com.linkedin.metadata.utils.SystemMetadataUtils;
+import com.linkedin.mxe.GenericAspect;
+import com.linkedin.mxe.SystemMetadata;
+import jakarta.json.Json;
+import jakarta.json.JsonPatch;
+import jakarta.json.JsonPatchBuilder;
+import java.io.StringReader;
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@RequiredArgsConstructor
+public class PatchEntitiesResolver
+    implements DataFetcher<CompletableFuture<List<PatchEntityResult>>> {
+
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+  private final EntityService _entityService;
+  private final EntityClient _entityClient;
+  private final EntityRegistry _entityRegistry;
+
+  @Override
+  public CompletableFuture<List<PatchEntityResult>> get(DataFetchingEnvironment environment)
+      throws Exception {
+    final List<PatchEntityInput> inputs =
+        bindArgument(environment.getArgument("input"), List.class);
+    final QueryContext context = environment.getContext();
+
+    return GraphQLConcurrencyUtils.supplyAsync(
+        () -> {
+          try {
+            return patchEntities(inputs, context);
+          } catch (Exception e) {
+            log.error("Failed to patch entities: {}", e.getMessage(), e);
+            return inputs.stream()
+                .map(input -> new PatchEntityResult(input.getUrn(), false, e.getMessage()))
+                .collect(Collectors.toList());
+          }
+        },
+        this.getClass().getSimpleName(),
+        "patchEntities");
+  }
+
+  private List<PatchEntityResult> patchEntities(
+      @Nonnull List<PatchEntityInput> inputs, @Nonnull QueryContext context) throws Exception {
+
+    final Authentication authentication = context.getAuthentication();
+    final List<Urn> entityUrns =
+        inputs.stream().map(input -> UrnUtils.getUrn(input.getUrn())).collect(Collectors.toList());
+
+    // Check authorization for all entities
+    if (!AuthUtil.isAPIAuthorizedEntityUrns(context.getOperationContext(), UPDATE, entityUrns)) {
```

Comment:
This is for API only meaning non-UI. There is a graphql util, something like AuthorizationUtil which is for graph. The reason is that historically the apis were not considering authorization while the graphql layer did. Because of that it is possible that graphql endpoints should be authorized but not APIs. This is no longer the default configuration but technically the isAPI is for OpenAPI and Rest.li whereas isAuthorized is for graphql. One day, probably when rest.li is removed we'd clean this all up.